openlibrary/macros/LoanStatus.html

```diff
@@ -15,6 +15,20 @@
 $ ocaid = doc.get('ocaid') or availability.get('identifier')
 $ work_key = work_key or (doc.get('works') and doc.works[0].key)
 
+$ book_provider = 'ia'
+
+$# Check for Project Gutenberg ID
+$ gutenberg_ids = doc.get('identifiers', {}).get('project_gutenberg') or doc.get('id_project_gutenberg')
+$ gutenberg_id = gutenberg_ids and gutenberg_ids[0]
+$if (gutenberg_id and (not ocaid or ocaid.endswith('gut'))):
+  $ book_provider = 'gutenberg'
+
+$# Check for LibriVox ID
+$ librivox_ids = doc.get('identifiers', {}).get('librivox') or doc.get('id_librivox')
+$ librivox_id = librivox_ids and librivox_ids[0]
+$if librivox_id:
+  $ book_provider = 'librivox'
+
```

Comment:
I think I might do this after the waiting_loan logic -- that block is if there's a book they currently have checked out / are reading which seems like a really strong intent.

Reply:
Oh, it is ever the `user_loan` block :+1:

Reply:
Oh, in the rendering part it's after. Here, these books are all PD, so there should never be a loan.

Reply:
Actually, I think I see what you mean; I'll re-arrange :+1: